hashintel/hash
# Fail tests on unexpected output count

## 🌟 What is the purpose of this PR?

When tests generate more outputs than the expected number we ignore the additional outputs. This PR makes it an error.

The default test output directory was defined using the `OUT_DIR` environment variable which requires a `build.rs`. This PR switches to `CARGO_TARGET_TMPDIR` which does not require a `build.rs` and is easier to access.

## 🔗 Related links

- [Asana task](https://app.asana.com/0/1199548034582004/1202206648648215/f) _(internal)_

## 🔍 What does this change?

- Change the default output folder for tests which allows us to remove the empty `build.rs`
- Make unexpected number of outputs an error instead of ignoring them

## 📜 Does this require a change to the docs?

No

## 🛡 What tests cover this?

All tests in `tests/examples`.

## ❓ How to test this?

1.  Checkout the branch / view the deployment
2.  Try `cargo test`
3.  Confirm that all tests pass
